src/Serilog.Sinks.EventLog/Sinks/EventLog/EventLogSink.cs

```diff
@@ -53,69 +58,153 @@ public EventLogSink(string source, string logName, ITextFormatter textFormatter,
+	                if (log == null)
+	                    return;
+
+	                var metaSource = $"serilog-{_logName}";
+	                if (!System.Diagnostics.EventLog.SourceExists(metaSource, machineName))
+	                    System.Diagnostics.EventLog.CreateEventSource(new EventSourceCreationData(metaSource, _logName)
+	                    {
+	                        MachineName = machineName
+	                    });
+
+	                log.Source = metaSource;
+	                log.WriteEntry(
+	                    message: $"Event source {source} was previously registered in log {existingLogWithSource.Log}. " +
+	                             $"The source has been registered with the new log, {logName}, however a computer restart is required " +
+	                             $"before event logs will appear in {logName} with source {source} instead of {existingLogWithSource.Log}.",
+	                    type: EventLogEntryType.Warning,
+	                    eventID: (int) LogEventLevel.Warning);
 	            }
 	        }
 	    }
 
 	    /// <summary>
-		/// Emit the provided log event to the sink.
-		/// </summary>
-		/// <param name="logEvent">The log event to write.</param>
-		/// <remarks>
-		/// <see cref="LogEventLevel.Debug" />, <see cref="LogEventLevel.Information" /> and <see cref="LogEventLevel.Verbose" /> are registered as <see cref="EventLogEntryType.Information" />.
-		/// <see cref="LogEventLevel.Error" />, <see cref="LogEventLevel.Fatal" /> are registered as <see cref="EventLogEntryType.Error" />.
-		/// <see cref="LogEventLevel.Warning" /> are registered as <see cref="EventLogEntryType.Warning" />.
-		/// The Event ID in the Windows log will be set to the integer value of the <paramref name="logEvent"/>'s <see cref="LogEvent.Level"/> property, so that the log can be filtered with more granularity.</remarks>
-		public void Emit(LogEvent logEvent)
-		{
-			if (logEvent == null) throw new ArgumentNullException("logEvent");
-
-			EventLogEntryType type;
-			switch (logEvent.Level)
-			{
-				case LogEventLevel.Debug:
-				case LogEventLevel.Information:
-				case LogEventLevel.Verbose:
-					type = EventLogEntryType.Information;
-					break;
-
-				case LogEventLevel.Error:
-				case LogEventLevel.Fatal:
-					type = EventLogEntryType.Error;
-					break;
-
-				case LogEventLevel.Warning:
-					type = EventLogEntryType.Warning;
-					break;
-
-				default:
-					SelfLog.WriteLine("Unexpected logging level, writing to EventLog as Information");
-					type = EventLogEntryType.Information;
-					break;
-			}
-
-			var payloadWriter = new StringWriter();
-			_textFormatter.Format(logEvent, payloadWriter);
-
-
-            //The payload is limitted in length and allowed chars
-            //see: https://msdn.microsoft.com/en-us/library/e29k5ebc%28v=vs.110%29.aspx
-            var payload = payloadWriter.ToString();
+        /// Returns name of EventLog that contains the registered source
```

Comment:
`*`shakes fist at VS`*` - worth dropping [editor.config](http://editorconfig.org/) config in here or just fix?